tiledb/api/src/array/mod.rs

```diff
@@ -380,6 +382,113 @@ impl Array {
         Ok(Some(datatype))
     }
 
+    pub fn vacuum<S>(
```

Comment:
New apis for vacuuming and consolidation

Reply:
It's a minor style choice but I feel a little bit like these should be module-level functions rather than part of the `impl`, @davisp do you have a preference?

Reply:
I don't have a preference in either direction. The only thought I have to push one way or the other is to check the C++ API which has them as static methods on Array as well:

https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/cpp_api/array.h#L588-L594

Reply:
If I understand correctly the inclination is then to leave them as impl (static)?

Reply:
per discussion, will leave as is